lang/java/reef-common/src/main/java/org/apache/reef/util/TangUtil.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.util;
+
+import org.apache.reef.tang.Configuration;
+import org.apache.reef.tang.Tang;
+import org.apache.reef.tang.annotations.Name;
+import org.apache.reef.tang.exceptions.InjectionException;
+
+/**
+ * A collection of helper functions to extract values from Tang configuration.
+ */
+public final class TangUtil {
+
+  private static final Tang TANG = Tang.Factory.getTang();
+
+  /**
+   * Do not instantiate this utility class - it's just a collection of static methods.
+   */
+  private TangUtil() { }
+
+  /**
+   * Extract a parameter's value from Tang configuration.
+   * @param config Tang configuration.
+   * @param clazz A parameter to extract from the configuration.
+   * @param defaultValue default value to use if the parameter is missing.
+   * @param <T> Type of the parameter - inferred from the parameter's class.
+   * @return Value of the configuration parameter, or the default value
+   * if it is not present in the Tang configuration.
+   */
+  public static <T> T getNamedInstance(
```

Comment:
Methods like this make me nervous, as they essentially negate the use of Tang and turn a `Configuration` instance into a mere hashmap. What is the reason for needing it? Maybe we can work on it at the callsite.